index.js

```diff
@@ -200,4 +200,10 @@ module.exports = function (options, callback) {
     matchCount++;
     callback(match, matchCount);
   }
+
+  function charIsEscaped(index) {
+    var esc = false;
+    while (source[--index] === "\\") esc = !esc;
```

Comment:
Can you say more about why we're ever pay attention to more than four slashes? In `\\\\\\"` (6 slashes), do those first two slashes have any impact at all on whether the `"` is escaped (i.e. is it the same as `\\\\\"` (5 slashes))?